### ISSUE TITLE
support metadataDevice for osd provisioning with ceph-volume

### ISSUE THREAD

**Issue by clkao** [rook/rook#2566]
When the cluster's ceph version supports ceph-volume, rook-ceph still uses traditional sgdisk-based provision for device when metadataDevice is specified: https://github.com/rook/rook/pull/2269/files#diff-d422456f63710f6140fbb62ba3b6ae8aR155 and there seems no followup issue on this.

Currently our use of `ceph-volume lvm batch` is per-device, as we need to support the osdsPerDevice param. this makes it difficult to make use of the `lvm batch` feature of mixed devices, where `ceph-volume` attempts to put db volume on ssd for multiple OSDs in one go.

```
[root@b4e442ada745 /]# ceph-volume lvm batch --prepare  --block-db-size 2048000000 /dev/sdb /dev/sdc /dev/sdd /dev/sde 

Total OSDs: 3

Solid State VG:
  Targets:   block.db                  Total size: 19.00 GB                 
  Total LVs: 3                         Size per LV: 1.91 GB                  
  Devices:   /dev/sdb

  Type            Path                                                    LV Size         % of device
----------------------------------------------------------------------------------------------------
  [data]          /dev/sdc                                                40.00 GB        100%
  [block.db]      vg: osd-db                                              1.91 GB         11%
----------------------------------------------------------------------------------------------------
  [data]          /dev/sdd                                                40.00 GB        100%
  [block.db]      vg: osd-db                                              1.91 GB         11%
----------------------------------------------------------------------------------------------------
  [data]          /dev/sde                                                40.00 GB        100%
  [block.db]      vg: osd-db                                              1.91 GB         11%
--> The above OSDs would be created if the operation continues
--> do you want to proceed? (yes/no) yes
```

But actually, if we simply pass the ssd metadata device to `lvm batch` runs, it will be used for block.db properly (vg is created the first time and will be reused)

In short, for metadataDevice to work with ceph-volume, The cheapest way seems to be making use of the existing lvm batch behaviour and pass the metadataDevice to it. Though there should probably be lots of checking to make sure `lvm batch` does what we intend to, as misconfigured ceph-volume params seem will likely trigger #2445 and is hard to diagnose.

**Comment by clkao** [comment]
Well it turns out there's already a placeholder for making use of `ceph-volume lvm batch` with multiple  devices. #2572 is a proof of concept bringing metadataDevice back to new OSDs provisioned with ceph-volume.
